vllm/entrypoints/llm.py

```diff
@@ -233,22 +233,29 @@ def _run_engine(self, use_tqdm: bool) -> List[RequestOutput]:
         # Initialize tqdm.
         if use_tqdm:
             num_requests = self.llm_engine.get_num_unfinished_requests()
-            pbar = tqdm(total=num_requests,
-                        desc="Processed prompts",
-                        dynamic_ncols=True)
+            pbar = tqdm(
+                total=num_requests,
+                desc="Processed prompts",
+                dynamic_ncols=True,
+                postfix=f"{0:.2f} toks/s",
+            )
         # Run the engine.
         outputs: List[RequestOutput] = []
+        total_tokens = 0
         while self.llm_engine.has_unfinished_requests():
             step_outputs = self.llm_engine.step()
             for output in step_outputs:
                 if output.finished:
                     outputs.append(output)
                     if use_tqdm:
+                        total_tokens += len(output.outputs[0].token_ids)
+                        speed = total_tokens / pbar.format_dict["elapsed"]
+                        pbar.postfix = f"{speed:.2f} toks/s"
```

Comment:
We should clarify this as decode/output/generation/generated tokens/s. This metrics does not capture the long prompt processing.